src/stactools/core/io/__init__.py

```diff
@@ -12,22 +12,23 @@
 
 
 def read_text(href: str,
-              read_href_modifier: Optional[ReadHrefModifier]) -> str:
+              read_href_modifier: Optional[ReadHrefModifier] = None) -> str:
     if read_href_modifier is None:
-        return pystac.STAC_IO.read_text(href)
+        return StacIO.default().read_text(href)
     else:
-        return pystac.STAC_IO.read_text(read_href_modifier(href))
+        return StacIO.default().read_text(read_href_modifier(href))
```

Comment:
Since this is only used in `stactools.core.io.xml` maybe it would be simpler/more efficient to create an instance of `StacIO.default()` in a class attribute on `XmlElement` and then just reference it in `XmlElement.from_file`?

```python
class XmlElement:

    stac_io: pystac.StacIO = StacIO.default()

    ...

    @classmethod
    def from_file(
        cls,
        href: str,
        read_href_modifier: Optional[ReadHrefModifier] = None
    ) -> "XmlElement":
        if read_href_modifier is not None:
            href = read_href_modifier(href)
        text = cls.stac_io.read_text(href)
        return cls(etree.fromstring(bytes(text, encoding='utf-8')))

```

Reply:
Could be, though the utility might be useful for anyone else using the read_href_modifier pattern elsewhere in the future. The StacIO.default() call should return fast after the first as it provides a cached value

Reply:
Yup, at least one downstream package is currently using (https://github.com/stactools-packages/landsat/blob/main/src/stactools/landsat/ang_metadata.py#L4).